lib/spack/spack/bootstrap.py

```diff
@@ -418,6 +418,16 @@ def _bootstrap_config_scopes():
 
 @contextlib.contextmanager
 def ensure_bootstrap_configuration():
+    # We may need to compile code from sources, so ensure we have compilers
+    # for the current platform before switching parts.
+    arch = spack.architecture.default_arch()
+    arch = spack.spec.ArchSpec(str(arch))  # The call below expects an ArchSpec object
+    if not spack.compilers.compilers_for_arch(arch):
+        compiler_cmd = spack.main.SpackCommand('compiler')
```

Comment:
are we using SpackCommand in our code directly now?

Reply:
Would rather not use it in the code, but in a pinch we can.  It would be better to factor this so that it does not go through `SpackCommand` IMO.

Reply:
We use `SpackCommand` in two places in `bootstrap.py`. The fact is our cli commands are not always thin wrappers of API calls but they may have a fair amount of logic in them. I agree this should be refactored, but I'd prefer to do it in a PR focused on refactoring the API - so that it wouldn't be out of place refactoring other related calls in an internal module and the PR would be focused on a single objective.